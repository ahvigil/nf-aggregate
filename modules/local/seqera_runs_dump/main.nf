process SEQERA_RUNS_DUMP {
    tag "$run_id"
    //secret 'TOWER_ACCESS_TOKEN'
    conda 'bioconda::tower-cli=0.9.0'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/tower-cli:0.9.0--h9ee0642_0' :
        'biocontainers/tower-cli:0.9.0--h9ee0642_0' }"

    input:
    val run_id
    val workspace

    output:
    tuple val(run_id), path("*_run_dump"), emit: run_dump
    path "versions.yml"                  , emit: versions

    script:
    def args = task.ext.args ?: ''
    def args2 = task.ext.args2 ?: ''
    def prefix = task.ext.prefix ?: "${run_id}"
    """
    tw \\
        --access-token=\$TOWER_ACCESS_TOKEN \\
        $args \\
        runs \\
        dump \\
        -id=$run_id \\
        --workspace=$workspace \\
        --output="${prefix}_run_dump.tar.gz" \\
        $args2

    mkdir ${prefix}_run_dump
    tar \\
        -xvf \\
        ${prefix}_run_dump.tar.gz \\
        -C ${prefix}_run_dump

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        seqera-cli: \$(echo \$(tw --version 2>&1) | sed 's/^.*Tower CLI version //; s/ *\$//')
    END_VERSIONS
    """
}
