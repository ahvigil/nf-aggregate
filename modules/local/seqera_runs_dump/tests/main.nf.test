nextflow_process {

    name "Test Process SEQERA_RUNS_DUMP"
    script "../main.nf"
    process "SEQERA_RUNS_DUMP"

    tag "modules"
    tag "modules_local"
    tag "seqera_runs_dump"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = ['id': '4Bi5xBK6E2Nbhj', 'workspace': 'community/showcase']
                """
            }
        }

        then {
            def dumpFiles = path(process.out.run_dump.get(0).get(1)).toFile().listFiles()

            assertAll(
                { assert snapshot(process.out.workflow_json).match() },
                { assert snapshot(workflow, dumpFiles.findAll { ! it.name.matches('workflow-launch.json')}).match()},
                { assert dumpFiles.find { it.name.matches('workflow-launch.json')}.exists()},
                { assert process.success }
            )
        }

    }

}
