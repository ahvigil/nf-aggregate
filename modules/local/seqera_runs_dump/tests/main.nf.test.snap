{
    "Should run without failures": {
        "content": [
            [
                "service-info.json:md5,0a2624c21efb65b85e37fa3d07903dcc",
                "workflow-load.json:md5,4f02d5a24ab89aa648cd4346785c8f2c",
                "workflow-metrics.json:md5,70dd1af37145c8c2d23836fe850622e7",
                "workflow-tasks.json:md5,577a7472816b7729012a9291d97ff150",
                "workflow.json:md5,34db0e39f0246670eac77c7a5e9093c9"
            ],
            null
        ],
        "timestamp": "2023-11-29T13:34:09.5038157"
    }
}